src/Hosting/Hosting/test/HostTests.cs

```diff
@@ -159,6 +159,27 @@ public async Task HostedServiceStartNotCalledIfHostNotStarted()
             }
         }
 
+        [Fact]
+        public async Task HostedServiceRegisteredAsSingletons()
+        {
+            using (var host = CreateBuilder()
+                   .ConfigureServices((hostContext, services) =>
+                   {
+                       services.AddHostedService<TestHostedService>();
+                   })
+                   .Build())
+            {
+                var svc = (TestHostedService)host.Services.GetRequiredService<IHostedService>();
```

Comment:
Call `GetService` at elast twice to prove the instance is indeed the same?

Reply:
+1 and ensure reference equality of the 2nd result

Reply:
Send a pull request?

Reply:
I have asked about it here: https://github.com/dotnet/extensions/pull/552#commitcomment-31388021
Answers are below.